### ISSUE TITLE
Top of Photon Tabs not clickable when using Compact Mode on FF 101+

### ISSUE THREAD

**Issue by jairrencora** [black7375/Firefox-UI-Fix#384]
**Describe the bug**
When using Compact Density and clicking on the top of a tab, instead of clicking the tab, the click event is relayed to the window. Previously it changed to the tab being clicked, that behavior is also still present on Normal Density. See video for more information.

**Expected behavior**
When clicking the top of the tab, Firefox should change to that tab

**Screenshots**
Please note, I don't know how to capture clicks visually, but while traversing through tabs I was clicking them to try to switch to them but as can be seen, nothing happens. I show in the recording the buggy behavior on Compact density and also the correct behavior on Normal Density. If there is a problem with the file host, let me know where it is appropriate to upload the recording.
https://we.tl/t-PdgJoLnffj

https://user-images.githubusercontent.com/25581533/166869151-b39a7ba3-bb46-441b-9a2b-74ca4f810b84.mp4

**Environment:**
 - Distribution
   - [ ] [Original Lepton](https://github.com/black7375/Firefox-UI-Fix)
   - [X] [Lepton's photon style](https://github.com/black7375/Firefox-UI-Fix/tree/photon-style)
   - [ ] [Lepton's proton style](https://github.com/black7375/Firefox-UI-Fix/tree/proton-style)
 - Firefox Version: 101.0.b2 and 102.0.a1
 - OS:
   - [ ] Linux
   - [X] Windows
   - [ ] Mac
   - [ ] Other: 
 - Theme:
   - [ ] Light
   - [X] Dark
   - [ ] Alpenglow (System: Light)
   - [ ] Alpenglow (System: Dark)
   - [ ] System Default
     - GTK Theme Name (Linux only): 
   - [ ] Other: (e.g. [Photon Colors](https://addons.mozilla.org/en-US/firefox/addon/photon-colors/), [Google Chrome Dark](https://addons.mozilla.org/en-US/firefox/addon/google-chrome-dark/), [Fairytale Of Nature](https://addons.mozilla.org/en-US/firefox/addon/fairytale-of-nature/))

**`user.js` setup**
Default for Lepton Photon Fix 5.0.2

**Additional context**
Bug only seems to happen on Compact Density and seems to have been caused by Firefox update 101. Hopefully the fix isn't outside of the scope of this project.

**Comment by tokariu** [comment]
it appears that this is the cause of the issue:
![appearance](https://user-images.githubusercontent.com/7195846/171386590-e5d600aa-d5e0-4f53-8c9b-81ea9346d0e6.png)

i fixed it for myself overwriting the code with:
```
:root[tabsintitlebar][sizemode="maximized"] #titlebar {
appearance: unset !important;
}
```
haven't tested for side effects, though. but this might be a fix.

**Comment by black7375** [comment]
Oh..Thank you I'll check it.
It's a bit tricky issue.

**Comment by jairrencora** [comment]
Tried @tokariu 's fix. It fixes the issue, but for me, it adds a thick dark border to all context menus. Also causes the screenshot function to display the selection grid completely dark. I'm using dark theme.

**Comment by tokariu** [comment]
@jairrencora I'm only using FF v101.0, I don't have these issues. Seems FF 102.0.b2 introduces new problems which I'll probably face first when v102 becomes live. Reading the upstream issues though, there might be several other issues coming with 102..

**Comment by jairrencora** [comment]
Yes @tokariu . Just saw this has been reported in Issue 401 as an issue in FF 102.0.b2 . It seems I added your fix at the same time my FF got updated and I mistakenly assumed it was due to the fix, when it was caused by the update. Disregard my previous comment.

**Comment by black7375** [comment]
@jairrencora Can I close this issue?

**Comment by UncleScrooge85** [comment]
Starting with FF 101 they have added a "window frame" bar on top of the tabs, thus slightly increasing the height that the top bars take on screen and reducing the available vertical space for web contents (again).

I suppose this issue is related to this change.
Is it possible to fix the UI to make it as it was before?

**Comment by phx7** [comment]
> it appears that this is the cause of the issue: ![appearance](https://user-images.githubusercontent.com/7195846/171386590-e5d600aa-d5e0-4f53-8c9b-81ea9346d0e6.png)
> 
> i fixed it for myself overwriting the code with:
> 
> ```
> :root[tabsintitlebar][sizemode="maximized"] #titlebar {
> appearance: unset !important;
> }
> ```
> 
> haven't tested for side effects, though. but this might be a fix.

Works just fine for me, big thanks. No issues found yet, context menus look unchanged.

**Comment by jairrencora** [comment]
@black7375 No, please. The original issue that I reported is still valid. Only my second comment about (what I thought were) side-effects from tokariu's fix is to be disregarded since it was related to issue Issue 401, but the original issue of the space between tabs and window in Compact mode is still present unless the fix is applied.

**Comment by black7375** [comment]
I tried to try VM because the windows computer was broken.

In general, @tokariu solution seems to work well.
Try to use the last commit.

**Comment by hiyokosan** [comment]
I reported is still valid

**Comment by hiyokosan** [comment]
Still seems the problem has not been resolved...

**Comment by black7375** [comment]
It seems to have been solved by the latest nightly.  #414
It will be changed to the compatibility option in the next release.

**Comment by duncanyoyo1** [comment]
This is still an issue for me on FF version 103.0.2, using latest master b7d3da3a0b57d0cf3350310a09022eab933098cc

Compact mode tabs are still not clickable from the top of the screen when maximized. Have to move down 2px. 

I tried @tokariu's fix but it didn't seem to work for me. I didn't notice any change.

**Comment by hiyokosan** [comment]
There is no entry for "userChrome.compatibility.os.windows_maximized" in the user.js file. 

After adding the above item.

 :root[tabsintitlebar][sizemode="maximized"] #titlebar { 
/* -moz-default-appearance: -moz-window-titlebar */
  appearance: none ! 
  padding-top: 0px; 

This solved the problem.

**Comment by black7375** [comment]
First of all, I will open the issue again.

**Comment by duncanyoyo1** [comment]
> ```css
>  :root[tabsintitlebar][sizemode="maximized"] #titlebar { 
> /* -moz-default-appearance: -moz-window-titlebar */
>   appearance: none ! 
>   padding-top: 0px; 
> }
> ```

> There is no entry for userChrome.compatibility.os.windows_maximized in the user.js file.

Mine certainly has a `userChrome.compatibility.os.windows_maximized` entry. Line 67. But it's commented out by default, and there's a spelling mistake.

It is currently
`user_pref("userChrome.compatibility.os.windows_maximized, true");`
instead of
`user_pref("userChrome.compatibility.os.windows_maximized", true);`

That said, even enabling it and adding the css you showed did not make the tabs clickable from the top of the screen. And it actually made the area above the tabs larger.

I ended up setting `margin-top: -8px;` and it's working for me now.

So to summarize what I did to get it working ( probably not correct, I am no CSS master ), 

1. Fix line 67 by uncommenting and moving the comma to the correct location.
2. Added the following to leptonChrome.css

> ```css
>  :root[tabsintitlebar][sizemode="maximized"] #titlebar { 
> /* -moz-default-appearance: -moz-window-titlebar */
>   appearance: none ! 
>   padding-top: 0px; 
>   margin-top: -8px;
> }
> ```

**Comment by hiyokosan** [comment]
Thanks for fixing user.js. 

Correction regarding the above css.
 It seems that the `"!important"` is needed instead of `"appearance: none;"`. 
Also, it seems that `"padding-top: -6px"` is not necessary.

**Comment by duncanyoyo1** [comment]
When I don't have the negative padding I get this,
![Screenshot 2022-08-29 033123](https://user-images.githubusercontent.com/24286951/187147972-06f317ed-42aa-4a55-8737-29d5fff15b6a.png)

Might be difficult to see, but there is extra space above the taps that is not clickable.

Not sure why this is the case as I haven't changed anything else.

**Comment by black7375** [comment]
@Urist0595 
What version of Firefox do you use?
It seems to work well in the latest version.

**Comment by Urist0595** [comment]
> @Urist0595 What version of Firefox do you use? It seems to work well in the latest version.

I'm using the latest version of Firefox. I was previously using the ESR version of Firefox but I switched to the latest version in hopes that this problem would stop happening, but it still is.

**Comment by hiyokosan** [comment]
Does it reproduce if I try a new profile?

**Comment by black7375** [comment]
It is not reproduced when you run `v102.6.0ESR` and `v108.0.1` versions in `Windows10`.

https://user-images.githubusercontent.com/25581533/208485478-1c930304-2075-4412-b770-abd28127c3ee.mp4

https://user-images.githubusercontent.com/25581533/208485601-91917e9f-8441-4ab2-8ef8-81aad4a5d6c0.mp4

---

It is possible to reproduce if there is more accurate information about `OS version` and `Firefox version`.

**Comment by Urist0595** [comment]
OS Version is Windows_NT 10.0 19045 and Firefox version is 108.0.1 according to the about:support page.

**Comment by Urist0595** [comment]
> Does it reproduce if I try a new profile?

I just tried making a new profile, and the issue did not reproduce. The top bar was clickable without me needing to edit leptonChrome.css like I always do. So for some reason this only happens on my default profile.

**Comment by black7375** [comment]
Is the `userChrome.compatibility.os.windows_maximized` option to `true` in `about:config`?
If it is `true`, try to modify it to `false`.

**Comment by Urist0595** [comment]
> Is the `userChrome.compatibility.os.windows_maximized` option to `true` in `about:config`? If it is `true`, try to modify it to `false`.

It was set to true, I just set it to false. After reinstalling Lepton on the default profile, the top bar was clickable. Thanks a lot, it looks like that was the issue.
